src/Nethermind/Nethermind.Baseline/JsonRpc/BaselineModule.cs

```diff
@@ -0,0 +1,307 @@
+
+            _logger.Info($"Sent transaction at price {tx.GasPrice} to {tx.SenderAddress}");
+            _logger.Info($"Contract {contractType} has been deployed");
+
+            return ResultWrapper<Keccak>.Success(txHash);
+        }
+
+        private BaselineTree RebuildEntireTree(Address treeAddress)
+        {
+            // bad
+
+            Keccak leavesTopic = new Keccak("0x8ec50f97970775682a68d3c6f9caedf60fd82448ea40706b8b65d6c03648b922");
+            LogFilter insertLeavesFilter = new LogFilter(
+                0,
+                new BlockParameter(0L),
+                new BlockParameter(_blockFinder.Head.Number),
+                new AddressFilter(treeAddress),
+                new TopicsFilter(new SpecificTopic(leavesTopic)));
+
+            Keccak leafTopic = new Keccak("0x6a82ba2aa1d2c039c41e6e2b5a5a1090d09906f060d32af9c1ac0beff7af75c0");
+            LogFilter insertLeafFilter = new LogFilter(
+                0,
+                new BlockParameter(0L),
+                new BlockParameter(_blockFinder.Head.Number),
+                new AddressFilter(treeAddress),
+                new TopicsFilter(new SpecificTopic(leafTopic))); // find tree topics
+
+            var insertLeavesLogs = _logFinder.FindLogs(insertLeavesFilter);
+            var insertLeafLogs = _logFinder.FindLogs(insertLeafFilter);
+            BaselineTree baselineTree = new ShaBaselineTree(new MemDb(), new byte[0], 5);
+
+            // Keccak leafTopic = new Keccak("0x8ec50f97970775682a68d3c6f9caedf60fd82448ea40706b8b65d6c03648b922");
+            foreach (FilterLog filterLog in insertLeavesLogs)
+            {
+                for (int i = 0; i < (filterLog.Data.Length - 128) / 32; i++)
+                {
+                    Bytes32 leafHash = Bytes32.Wrap(filterLog.Data.Slice(128 + 32 * i, 32).ToArray());
+                    baselineTree.Insert(leafHash);
+                }
+            }
+
+            foreach (FilterLog filterLog in insertLeafLogs)
+            {
+                Bytes32 leafHash = Bytes32.Wrap(filterLog.Data.Slice(32, 32).ToArray());
+                baselineTree.Insert(leafHash);
+            }
+
+            return baselineTree;
+        }
+
+        public Task<ResultWrapper<BaselineTreeNode[]>> baseline_getSiblings(Address contractAddress, long leafIndex)
+        {
+            if (leafIndex > MerkleTree.MaxLeafIndex || leafIndex < 0L)
+            {
+                return Task.FromResult(ResultWrapper<BaselineTreeNode[]>.Fail($"{leafIndex} is not a valid leaf index", ErrorCodes.InvalidInput));
+            }
+
+            bool isTracked = _baselineTrees.TryGetValue(contractAddress, out BaselineTree? tree);
+            if (!isTracked)
+            {
+                return Task.FromResult(ResultWrapper<BaselineTreeNode[]>.Fail($"{contractAddress} tree is not tracked", ErrorCodes.InvalidInput));
+            }
+
+            // everything in memory
+            tree = RebuildEntireTree(contractAddress);
```

Comment:
Why do we have to rebuilt tree?

Reply:
we do not have to - I left the DB building and change / reorg tracking code for later so we can test this PoC